presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoDatabaseMetaData.java

```diff
@@ -1499,4 +1514,14 @@ private static void quoteStringLiteral(StringBuilder out, String value)
         }
         out.append('\'');
     }
+
+    private boolean stringIsLikePattern(String value) throws SQLException
+    {
+        if (value == null) {
+            return false;
+        }
+        Pattern p = Pattern.compile("[%_]");
```

Comment:
I'm not sure this can be done correctly with a regex. There are two choices:

1. Don't worry about the escape character and assume anything with pattern characters is potentially a pattern:
```java
return value.contains("%") || value.contains("_");
```
2. Use a simple state machine to handle escapes, using the same algorithm as `LikeFunctions.likePattern()`:
```java
boolean escaped = false;
for (char currentChar : patternString.toCharArray()) {
    if (!escaped && (currentChar == '\\')) {
        escaped = true;
    }
    else if (escaped) {
        escaped = false;
    }
    else if ((currentChar == '%') || (currentChar == '_')) {
        return true;
    }
}
return false;
```
Make the method or the callers `@VisibleForTesting` and test with some strings like (these are exact strings not Java source code strings):

* `x%x` -- is a pattern
* `x\\%x` -- is a pattern since the `\` is escaped
* `x\\\%x` -- not a pattern